launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java

```diff
@@ -143,59 +143,61 @@ static List<String> parseOptionString(String s) {
         escapeNext = false;
       } else if (inOpt) {
         switch (c) {
-        case '\\':
-          if (inSingleQuote) {
-            opt.appendCodePoint(c);
-          } else {
-            escapeNext = true;
+          case '\\' -> {
```

Comment:
For this particular case, I personally believe that `Pattern Matching for switch` does not demonstrate its advantages.

Reply:
Yes. This case can't be simplified. But it is safer than before.

Reply:
OK